Services/Interfaces/web_interface/static/js/common/resources_rendering.js

```diff
@@ -38,23 +38,99 @@ function fetch_images() {
+}
+
+
+function fetchCurrencyImage(element, currencyId){
+    if(element.parents("#AddCurrency-template-default").length > 0){
+        // do not fetch images for default elements
+        return;
+    }
+    if (fetchingCurrencies.indexOf(currencyId) === -1){
+        fetchingCurrencies.push(currencyId);
+        $.get({
+            url: `https://api.coingecko.com/api/v3/coins/${currencyId}?localization=false&tickers=false&market_data=false&community_data=false&developer_data=false&sparkline=false`,
+            dataType: "json",
+            success: function(data){
+                if(isDefined(data["image"])){
+                    element.attr("src", data["image"]["large"]);
+                    fetchingCurrencies.splice(fetchingCurrencies.indexOf(currencyId), 1);
+                }else{
+                    _useDefaultImage(element, currencyId);
+                }
+            },
+            error: function(result, status){
+                window.console&&console.error(`Impossible to get the currency image for ${currencyId}: ${result.responseText} (${status})`);
+                _useDefaultImage(element, currencyId);
+            }
+        });
+    }
+}
+
+function fetchCurrencyIds(){
+    currencyIdBySymbol = {};
+    $.get({
+        url: currencyListURL,
+        dataType: "json",
+        success: function (data) {
+            $.each(data, function (_, element){
+                currencyIdBySymbol[element["symbol"].toLowerCase()] = element["id"];
+            });
+            fetchedCurrencyIds = true;
+            // refresh images
+            handleDefaultImages();
+        },
+        error: function (result, status) {
+            window.console && console.error(`Impossible to get currency list from coingecko.com: ${result.responseText} (${status})`);
+        }
+    });
+}
+
 function handleDefaultImages(){
     $(".currency-image").each(function () {
-        handleDefaultImage($(this), currencyDefaultImage);
+        const element = $(this);
+        const imgSrc = element.attr("src");
+        if (imgSrc === "" || imgSrc.endsWith(currencyLoadingImageName)) {
+            if (element[0].hasAttribute("data-currency-id")) {
+                fetchCurrencyImage(element, element.attr("data-currency-id").toLowerCase());
+            }else if (element[0].hasAttribute("symbol")){
+                const symbol = element.attr("symbol").toLowerCase();
+                if (typeof currencyIdBySymbol === "undefined"){
+                    fetchCurrencyIds();
+                }else if (fetchedCurrencyIds){
+                    if (currencyIdBySymbol.hasOwnProperty(symbol)){
+                        fetchCurrencyImage(element, currencyIdBySymbol[symbol]);
+                    }else{
+                        handleDefaultImage(element, currencyDefaultImage);
+                    }
+                }
+            }
+        }
     });
 }
 
+const currencyLoadingImageName = "loading_currency.svg";
```

Comment:
Is it still the same default currency image ?

Reply:
It's this 
![image](https://user-images.githubusercontent.com/9078616/99192185-cd0e8280-2771-11eb-8303-8b3fab02c8db.png)